lib/options.go

```diff
@@ -164,6 +175,26 @@ func (c *TLSAuth) Certificate() (*tls.Certificate, error) {
 	return c.certificate, nil
 }
 
+func parsePrivateKey(privKey, password string) ([]byte, error) {
+	key := []byte(privKey)
+	block, _ := pem.Decode(key)
+	if block == nil {
+		return nil, fmt.Errorf("failed to decode PEM key")
+	}
+	blockType := block.Type
+	if encrypted := x509.IsEncryptedPEMBlock(block); encrypted {
```

Comment:
I assumed that some users might provide it by mistake but I also understand assuming that they do not.

Reply:
I guess that you can't log here :thinking: . But on the other hand I would argue it should be an outright error if there *is a* password, and it isn't encrypted. If you have a password and not DEK header ... you maybe shouldn't have `password` ?

Reply:
Yes, I understand your point. I just thought it confuse some users the first time they use the feature.

Reply:
>  confuse some users

I would argue it will be even more confusing if you add an option and nothing changes - if we at least then error out differently IMO it's better.